### PR TITLE
create Balancer trades on Dune V2

### PR DESCRIPTION
**Pull request by mendesfabio**
Brief comments on the purpose of your changes:

*For Dune Engine V2*
I've checked that:
General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)
* [x] if adding a new materialized table, I edited the optimize table macro

Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by jeff-dude** [comment]
> should I add `balancer_trades` tables here as well? https://github.com/duneanalytics/spellbook/blob/main/macros/optimize_tables.sql

only objects materialized as tables, i.e. ```balancer_v1_ethereum.trades``` and ```balancer_v2_ethereum.trades```

**Comment by mendesfabio** [comment]
done, thanks @jeff-dude

**Comment by jeff-dude** [comment]
@mendesfabio please merge our ```main``` into your branch again, you should see the optimize macro be deleted (it's handled in a global macro now, no need to reconfigure)

File: models/balancer/ethereum/balancer_ethereum_schema.yml [hunk 1]
```diff
@@ -1,13 +1,187 @@
...
+        description: "Raw value of the token sold at time of execution in the original currency"
+      - &amount_usd
+        name: amount_usd
+        description: "USD value of the trade at time of execution"
+      - &token_bought_address
+        name: token_bought_address
+        description: "Contract address of the token bought"
+        tests:
+          - dex_trades_token_bought:
+              dex_trades_seed: ref('dex_trades_seed')
```
Comment (jeff-dude) [inline models/balancer/ethereum/balancer_ethereum_schema.yml:69]: in order for this test to run, you need to manually add rows to the seed file her for balancer (both v1 and v2):
https://github.com/duneanalytics/spellbook/blob/main/seeds/dex/trades/dex_trades_seed.csv

when i added for uniswap, i went to their app & opened their analytics page, found some tx's links to etherscan, grabbed all relevant info and added to seed file.

hopefully that's easy to do with balancer as well (for v1 and v2)

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Comment by jeff-dude** [comment]
please also note this commit where i pushed some changes to help finalize:
https://github.com/duneanalytics/spellbook/pull/1642/commits/6c18bb665c6cd1755e6d1c608c83c1d679d8fbab

**Comment by jeff-dude** [comment]
closing due to inactivity, please do open when ready and we'll be happy to help finalize 🙏

**Pull request closed without merging**